viewer/src/public/migration/Cognite3DViewer.ts

```diff
@@ -1042,6 +1045,82 @@ export class Cognite3DViewer {
     throw new NotSupportedInMigrationWrapperError('Cache is not supported');
   }
 
+  /**
+   * Creates a tool for attaching HTML elements to a 3D position and updates it's
+   * position/visibility as user moves the camera. This is useful to create HTML
+   * overlays to highlight information about key positions in the 3D model.
+   *
+   * Attached elements *must* have CSS style 'position: absolute'. It's also recommended
+   * in most cases to have styles 'pointerEvents: none' and 'touchAction: none' to avoid
+   * interfering with 3D navigation. Consider also applying 'transform: translate(-50%, -50%)'
+   * to anchor the center of the element rather than the top-left corner. In some cases the
+   * `zIndex`-attribute is necessary for the element to appear on top of the viewer.
+   *
+   * @example
+   * ```js
+   * const el = document.createElement('div');
+   * el.style.position = 'absolute'; // Required!
+   * // Anchor to center of element
+   * el.style.transform = 'translate(-50%, -50%)';
+   * // Avoid being target for events
+   * el.style.pointerEvents = 'none;
+   * el.style.touchAction = 'none';
+   *
+   * el.style.color = 'red';
+   * el.innerHtml = '<h1>Overlay</h1>';
+   *
+   * const overlayTool = viewer.createHtmlOverlays();
+   * overlayTool.attachHtmlElement(el, new THREE.Vector3(10, 10, 10));
+   * // ...
+   * overlayTool.removeHtmlElement(el);
+   * // or, to remove all attached elements:
+   * overlayTool.dispose();
+   * ```
+   */
+  createHtmlOverlayTool(): HtmlOverlayTool {
+    const overlays = new HtmlOverlayTool(this.domElement, this.renderer, this.camera);
+    this.registerTool(overlays);
+    return overlays;
+  }
+
+  /**
+   * Registers a new tool. This makes the tool get updates from the viewer.
+   * The tool will be unregistered when it's diposed.
+   * @param tool Tool to register.
+   */
+  private registerTool(tool: Cognite3DViewerTool): void {
+    if (this._tools.has(tool)) {
+      throw new Error(`Tool ${tool} is already registered`);
+    }
+    const onDisposed: DisposedDelegate = () => this.unregisterTool(tool);
+    this._tools.set(tool, onDisposed);
+    tool.on('disposed', onDisposed);
+  }
+
+  /**
+   * Unregisters a previously registered tool for updates. Note! Tools
+   * are automatically unregistered when they are disposed so you will
+   * probably not need to call this function.
+   * @param tool A previously registered tool.
+   */
+  private unregisterTool(tool: Cognite3DViewerTool): void {
+    const onDisposed = this._tools.get(tool);
+    if (onDisposed === undefined) {
+      throw new Error(`Tool ${tool} is not registered`);
+    }
+    tool.off('disposed', onDisposed);
+    this._tools.delete(tool);
+  }
+
+  /**
+   * Notifies all registered tools that there has been rendered a frame.
+   */
+  private notifyToolsAboutRendering() {
```

Comment:
As a second thought on the API, does it make sense for viewer to know about tools in that way? I mean, maybe instead tools should know about the viewer and subscribe on its 'rendered' or 'disposed' or other events?

Reply:
This has now been reversed - tools know about viewer, not the other way around.